### ISSUE TITLE
Enhance aws_s3_bucket lifecycle to support tag filter.

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#675]
_This issue was originally opened by @djrichar as hashicorp/terraform#13473. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

please enhance s3 bucket lifecycle to support the new tags filter

### Terraform Version
Terraform v0.9.2

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_s3_bucket

### Suggested Terraform Configuration
```hcl
resource "aws_s3_bucket" "bucketname" {
  bucket = "bucketname"
  lifecycle_rule {
    id      = "delete 30 days when tag DELETE = 30"
    prefix  = ""
    tags    = { 
      DELETE = 30
    }
    enabled = true

    expiration {
      days = 30
    }
  }
```

### Expected Behavior
What should have happened?
s3 bucket should have a lifecycle that deletes all objects with a tage 'DELETE' and value '30'

### Actual Behavior
What actually happened?
NOT SUPPORTED

### References
see http://docs.aws.amazon.com/AmazonS3/latest/dev/intro-lifecycle-rules.html#intro-lifecycle-rules-filter  for documentation

**Comment by bdashrad** [comment]
hashicorp/terraform#13680 would fix this.

**Comment by bdashrad** [comment]
New PR for fixing this: terraform-providers/terraform-provider-aws#899

**Comment by bdashrad** [comment]
This has been fixed since version 0.1.2 / #899.

**Comment by Ninir** [comment]
Thanks for the ping @bdashrad !
